src/test/java/org/broadinstitute/consent/http/cloudstore/GCSServiceTest.java

```diff
@@ -139,4 +140,30 @@ void testDeleteDocument() {
     assertTrue(deleted);
   }
 
+  @Test
+  void testReadJsonFileFromBucket() throws Exception {
+    String testJson = """
+        {
+          "institutionDomainMap": {
+            "Broad Institute": ["broadinstitute.org", "broad.mit.edu"],
+            "Harvard": ["harvard.edu", "hms.harvard.edu"]
+          }
+        }
+        """;
+
+    String blobIdName = "institution-domain-mapping.json";
+    Blob blob = mock(Blob.class);
+    when(blob.getContent()).thenReturn(testJson.getBytes());
+    when(storage.get(any(BlobId.class))).thenReturn(blob);
+
+    initStore();
+
+    InstitutionDomainMap result = service.readJsonFileFromBucket(blobIdName,
+        InstitutionDomainMap.class);
+
+    assertNotNull(result);
+    assertEquals(2, result.getInstitutionDomainMap().size());
+    assertTrue(
+        result.getInstitutionDomainMap().get("Broad Institute").contains("broadinstitute.org"));
```

Comment:
You could use hamcrest here
```suggestion
    assertThat(result.getInstitutionDomainMap(),
        allOf(aMapWithSize(2), hasEntry("Broad Institute", Set.of("broadinstitute.org", "broad.mit.edu"))));
```

Reply:
The rest of the code uses the `junit` Assertions, so I will use the same style.